process/accountProcessor.go

```diff
@@ -40,44 +40,52 @@ func NewAccountProcessor(proc Processor, pubKeyConverter core.PubkeyConverter, c
 }
 
 // GetShardIDForAddress resolves the request by returning the shard ID for a given address for the current proxy's configuration
-func (ap *AccountProcessor) GetShardIDForAddress(address string) (uint32, error) {
+func (ap *AccountProcessor) GetShardIDForAddress(address string) (uint32, int, error) {
 	addressBytes, err := ap.pubKeyConverter.Decode(address)
 	if err != nil {
-		return 0, err
+		return 0, http.StatusBadRequest, err
 	}
 
-	return ap.proc.ComputeShardId(addressBytes)
+	shardID, err := ap.proc.ComputeShardId(addressBytes)
+	if err != nil {
+		return 0, http.StatusInternalServerError, err
+	}
+
+	return shardID, http.StatusOK, nil
 }
 
 // GetAccount resolves the request by sending the request to the right observer and replies back the answer
-func (ap *AccountProcessor) GetAccount(address string) (*data.Account, error) {
-	observers, err := ap.getObserversForAddress(address)
+func (ap *AccountProcessor) GetAccount(address string) (*data.Account, int, error) {
+	observers, status, err := ap.getObserversForAddress(address)
 	if err != nil {
-		return nil, err
+		return nil, status, err
 	}
 
+	gHttpStatus := http.StatusInternalServerError
 	for _, observer := range observers {
 		responseAccount := &data.AccountApiResponse{}
 
-		_, err = ap.proc.CallGetRestEndPoint(observer.Address, AddressPath+address, responseAccount)
+		httpStatus, err := ap.proc.CallGetRestEndPoint(observer.Address, AddressPath+address, responseAccount)
 		if err == nil {
 			log.Info("account request", "address", address, "shard ID", observer.ShardId, "observer", observer.Address)
-			return &responseAccount.Data.AccountData, nil
+			return &responseAccount.Data.AccountData, http.StatusOK, nil
 		}
 
 		log.Error("account request", "observer", observer.Address, "address", address, "error", err.Error())
+		gHttpStatus = httpStatus
 	}
 
-	return nil, ErrSendingRequest
+	return nil, gHttpStatus, ErrSendingRequest
```

Comment:
I don't think it is correct. From what I see, you will return the latest http status.
Imagine that all observers are down. Then here, you will return `404`. When parsing the code, 400 is between [400, 500) so you will return `bad request` but it's not the caller's fault so I would rather return `InternalServerError` directly here.